osdep/semaphore_osx.c

```diff
@@ -67,32 +69,18 @@ int mp_sem_trywait(mp_sem_t *sem)
     return r;
 }
 
-int mp_sem_timedwait(mp_sem_t *sem, const struct timespec *abs_timeout)
+int mp_sem_timedwait(mp_sem_t *sem, int64_t until)
 {
     while (1) {
         if (!mp_sem_trywait(sem))
             return 0;
 
-        int timeout_ms = -1;
-        if (abs_timeout) {
-            timeout_ms = 0;
-
-            // OSX does not provide clock_gettime() either.
-            struct timeval tv;
-            gettimeofday(&tv, NULL);
-
-            if (abs_timeout->tv_sec >= tv.tv_sec) {
-                long long msec = (abs_timeout->tv_sec - tv.tv_sec) * 1000LL +
-                    abs_timeout->tv_nsec / 1000LL / 1000LL - tv.tv_usec / 1000LL;
-                if (msec > INT_MAX)
-                    msec = INT_MAX;
-                if (msec < 0)
-                    msec = 0;
-                timeout_ms = msec;
-            }
-        }
+        int64_t timeout = MPCLAMP(until - mp_time_ns(),
```

Comment:
Ahem... I'd hoist out the function call to avoid double (or triple?) evaluation.

Reply:
changed